### PR TITLE
Add plugin: Blog AI Generator

### PR DESCRIPTION
**Pull request by garethng**
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin:
[https://github.com/garethng/obsidian-blog-generator](https://github.com/garethng/obsidian-blog-generator)
## Release Checklist
- [ ] I have tested the plugin on
  - [x]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

### CODE REVIEW

**Comment by garethng** [comment]
fix

**Comment by Zachatoo** [comment]
[# Blog Generator for Obsidian](https://github.com/garethng/obsidian-blog-generator/blob/379c14521ccc5dc9f786b956148393152e725e23/README.md#L1), [Blog Generator for Obsidian](https://github.com/garethng/obsidian-blog-generator/blob/379c14521ccc5dc9f786b956148393152e725e23/README.md#L7), [# Obsidian 博客生成器](https://github.com/garethng/obsidian-blog-generator/blob/379c14521ccc5dc9f786b956148393152e725e23/README_zh-CN.md#L1), [# Obsidian 블로그 생성기](https://github.com/garethng/obsidian-blog-generator/blob/379c14521ccc5dc9f786b956148393152e725e23/README_ko.md#L1), [# Obsidian ブログジェネレーター](https://github.com/garethng/obsidian-blog-generator/blob/379c14521ccc5dc9f786b956148393152e725e23/README_ja.md#L1)
Naming something "Obsidian xyz" or "xyz for Obsidian" is reserved for first party products we create, please change this header.

[export const translations: Translations = {](https://github.com/garethng/obsidian-blog-generator/blob/379c14521ccc5dc9f786b956148393152e725e23/src/i18n.ts#L47-L220)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI), for example "Generate blog", "API base URL", "Use custom prompt", etc.

[import * as path from 'path';](https://github.com/garethng/obsidian-blog-generator/blob/379c14521ccc5dc9f786b956148393152e725e23/src/main.ts#L4)
This import is only available from a Node.js runtime, which means that this will throw errors on Mobile (that's a problem since you have `isDesktopOnly` marked as `false` in your manifest.json file.

[this.app.workspace.getLeavesOfType(VIEW_TYPE_BLOG_GENERATOR).forEach(leaf => leaf.detach());](https://github.com/garethng/obsidian-blog-generator/blob/379c14521ccc5dc9f786b956148393152e725e23/src/main.ts#L137)
Do not detach leaves with your custom view in `onunload`, this is an antipattern, see: <https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Don't+detach+leaves+in+%60onunload%60>

[private async findImageInFolder(imageName: string, folder: TFolder | null): Promise<TFile | null> {](https://github.com/garethng/obsidian-blog-generator/blob/379c14521ccc5dc9f786b956148393152e725e23/src/main.ts#L325-L353)
This is going to capture all embeds, not just image embeds, see <https://help.obsidian.md/Linking+notes+and+files/Embed+files>. I recommend getting image embeds from [`MetadataCache.getFileCache`](https://docs.obsidian.md/Reference/TypeScript+API/MetadataCache/getFileCache) instead.
```ts
const embeds = this.app.metadataCache.getFileCache(view.file);
const images = embeds
    .map(embed => this.app.metadataCache.getFirstLinkpathDest(embed.link, view.file))
    .filter(file => file.extension !== 'md'); // or filter for specific extensions, such as .png, .jpg, etc
```

[private async findImageInFolder(imageName: string, folder: TFolder | null): Promise<TFile | null> {](https://github.com/garethng/obsidian-blog-generator/blob/379c14521ccc5dc9f786b956148393152e725e23/src/main.ts#L325-L353)
[Avoid iterating all files to find a file by its path](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Avoid+iterating+all+files+to+find+a+file+by+its+path).

[console.log(this.t('FOUND_IMAGES'), images.length);](https://github.com/garethng/obsidian-blog-generator/blob/379c14521ccc5dc9f786b956148393152e725e23/src/main.ts#L182), [console.log(this.t('CALLING_API'));](https://github.com/garethng/obsidian-blog-generator/blob/379c14521ccc5dc9f786b956148393152e725e23/src/main.ts#L216), [console.log(this.t('API_CALL_SUCCESS'));](https://github.com/garethng/obsidian-blog-generator/blob/379c14521ccc5dc9f786b956148393152e725e23/src/main.ts#L233), [console.log(this.t('BLOG_GENERATION_COMPLETE'));](https://github.com/garethng/obsidian-blog-generator/blob/379c14521ccc5dc9f786b956148393152e725e23/src/main.ts#L271)
Please avoid unnecessary logging. If you need it for debugging purposes, add a check to only log during development.

[new Setting(containerEl)](https://github.com/garethng/obsidian-blog-generator/blob/379c14521ccc5dc9f786b956148393152e725e23/src/main.ts#L456-L467)
This setting isn't relevant if the "useCustomPrompt" setting is disabled. Please do not show this setting unless "useCustomPrompt" is enabled.

**Comment by garethng** [comment]
Hi @Zachatoo ,
Thanks for your reviewing. I have fix all the issues you mentioned. Please help review it again. Really thanks for your work.

BR
Gareth

**Comment by Zachatoo** [comment]
Your plugin should work fine on mobile if you remove the path import, you aren't using it.

**Comment by garethng** [comment]
Hi @Zachatoo ,
Please help review it again that I have removed the unused path import as suggested. Thanks for your work.

BR
Gareth

**Comment by joethei** [comment]
[api](https://github.com/garethng/obsidian-blog-generator/blob/72f551f96ec65b20b0bcb189110526e492788994/manifest.json#L6)
API is always capitalized.

[private readonly DEBUG = process.env.NODE_ENV !== 'production';](https://github.com/garethng/obsidian-blog-generator/blob/72f551f96ec65b20b0bcb189110526e492788994/src/main.ts#L71)
This will break on mobile, because `process` only exists on desktop.
You can configure a esbuild replacement to fix this, without touching this code.

[new Setting(containerEl)](https://github.com/garethng/obsidian-blog-generator/blob/72f551f96ec65b20b0bcb189110526e492788994/src/main.ts#L401-L415)
This should not be a setting, use `moment.locale()` to get the configured language from Obsidian.

**Comment by garethng** [comment]
Hi @joethei 
I have make some changes in my code to fix all of the three issues above. Please help review it again. Thanks for help.

BR
Gareth
